pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -208,11 +208,13 @@ def on_save_checkpoint(self, trainer, pl_module) -> Dict[str, Any]:
             "best_model_score": self.best_model_score,
             "best_model_path": self.best_model_path,
             "current_score": self.current_score,
+            "dirpath": self.dirpath
         }
 
     def on_load_checkpoint(self, checkpointed_state: Dict[str, Any]):
         self.best_model_score = checkpointed_state["best_model_score"]
         self.best_model_path = checkpointed_state["best_model_path"]
+        self.dirpath = checkpointed_state.get("dirpath", self.dirpath)
```

Comment:
also, I don't see it being used anywhere.

Reply:
Great question ! I had a chat with Adrian. It is more a design choice. Let's say we want to fine-tune a model from a given checkpoint. It would make sense for the new checkpoint to be saved in the same folder. Happy to brainstorm on this one.

Reply:
I think there are more things we need to restore like `save_top_k`, `best_k_models`, etc. So I'd suggest to keep it for another PR. Also there is an open issue for the same.